pkg/services/navtree/navtreeimpl/admin.go

```diff
@@ -84,8 +84,10 @@ func (s *ServiceImpl) getOrgAdminNode(c *contextmodel.ReqContext) (*navtree.NavL
 	if err != nil {
 		return nil, err
 	}
+	globalhideApiKeys, _, _ := s.kvStore.Get(c.Req.Context(), serviceaccounts.GloballyHideAPIKeysTabOrgID, "serviceaccounts", "hideApiKeys")
```

Comment:
Is there a better way we can solve this? atm we are doing tree db queries to see if we should hide the navigation on every hard reload

Reply:
currently considering if we could save it with different values. il revisit this

Reply:
made the count api keys method available. depending on what we decide for option, we can remove both of the other db calls

Reply:
Nice, that is an improvment. Now if we want to check in other places if can use api keys this part is a bit much:
```go
hideApiKeys, _, _ := s.kvStore.Get(c.Req.Context(), c.OrgID, "serviceaccounts", "hideApiKeys")
apiKeys, err := s.apiKeyService.CountAPIKeys(c.Req.Context(), c.OrgID)
if err != nil {
    return nil, err
}
globalhideApiKeys, _, _ := s.kvStore.Get(c.Req.Context(), serviceaccounts.GloballyHideAPIKeysTabOrgID, "serviceaccounts", "hideApiKeys")

// Hide API keys if the global setting is set or if the org setting is set and there are no API keys
apiKeysHidden := (globalhideApiKeys == "1" && apiKeys == 0) || (hideApiKeys == "1" && apiKeys == 0)
```
So instead could we hide this logic inside a function?

Reply:
Yea agree. could be made smoother. We have a meeting today to discuss what route we want to take on this. So I am holding off any development for now til after meeting with Vesna and Daniel.

Reply:
from our discussion we have decided to make an abstraction for this to make it more readable for the user. See followup PR - https://github.com/grafana/grafana/pull/64018/files#r1122974718